eng/pipelines/arcade/stage-unit-tests.yml

```diff
@@ -0,0 +1,170 @@
+parameters:
+  jobMatrix: []  # Should define jobMatrix.name, jobMatrix.displayName, jobMatrix.pool, jobMatrix.timeout, jobMatrix.testOS
+  mauiSourcePath: $(Build.SourcesDirectory)
+  buildConfig: Debug
+  repoLogPath: $(Build.Arcade.LogsPath)
+  publishTaskPrefix: '1ES.'
+
+stages:
+- stage: UnitTests
+  displayName: Run Unit Tests
+  dependsOn: []
+  jobs:
+  - ${{ each job in parameters.jobMatrix }}:
+    - job: ${{ job.name }}
+      displayName: ${{ job.displayName }}
+      pool: ${{ job.pool }}
+      timeoutInMinutes: ${{ job.timeout }}
+      steps:
+      - template: /eng/pipelines/arcade/setup-test-env.yml
+        parameters:
+          mauiSourcePath: ${{ parameters.mauiSourcePath }}
+          buildConfig: ${{ parameters.buildConfig }}
+          repoLogPath: ${{ parameters.repoLogPath }}
+          installPackageArtifacts: false
+
+      - template: /eng/pipelines/common/run-dotnet-preview.yml
+        parameters:
+          displayName: Run Controls.Core.UnitTests - ${{ job.testOS }}
+          mauiSourcePath: ${{ parameters.mauiSourcePath }}
+          command: test
+          project: ${{ parameters.mauiSourcePath }}/src/Controls/tests/Core.UnitTests/Controls.Core.UnitTests.csproj
+          arguments: '-c ${{ parameters.buildConfig }} --logger trx --results-directory $(Agent.TempDirectory)/Controls.Core.UnitTests'
+          useExitCodeForErrors: true
+
+      - task: PublishTestResults@2
+        inputs:
+          testResultsFormat: VSTest
+          testResultsFiles: $(Agent.TempDirectory)/Controls.Core.UnitTests/*.trx
+          testRunTitle: Controls.Core.UnitTests - ${{ job.testOS }}
+
+      - template: /eng/pipelines/common/run-dotnet-preview.yml
```

Comment:
yap, one of the things it does is run the toolset restore, that will run this project first 
https://github.com/dotnet/maui/blob/main/eng/Tools.props#L8 this will install .net by default and then run this targets that I use to install the workloads. 
So we can skip the run-dotnet-preview.yml and just call the script and it should find the correct .net to use. also like I said will make us easy to reproduce locally

Reply:
Should I try to address this now or look to use that build script in a follow up PR?

Reply:
I think we can do it in the next PR.